Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -688,11 +690,48 @@ namespace AZ::Reflection
+                        //      1. AZ::Crc32 - This is the default
+                        //      2. AZ::u32 - This allows the user to specify a value of 1/0 for Show/Hide, respectively
+                        //      3. bool - This allows the user to specify true/false for Show/Hide, respectively
+                        //
+                        // We need to return out of checkAttribute for Visibility attributes since the attributeValue handling
+                        // below doesn't account for these special cases. The Visibility attribute instead gets cached at
+                        // the end of the CacheAttributes method after it has done further visibility computations.
                         if (name == PropertyEditor::Visibility.GetName())
                         {
-                            visibility = PropertyEditor::Visibility
-                                             .DomToValue(PropertyEditor::Visibility.LegacyAttributeToDomValue(instance, it->second))
-                                             .value_or(visibility);
+                            auto visibilityValue = PropertyEditor::Visibility.DomToValue(
+                                PropertyEditor::Visibility.LegacyAttributeToDomValue(instance, it->second));
+
+                            if (visibilityValue.has_value())
+                            {
+                                visibility = visibilityValue.value();
+
+                                // The PropertyEditor::Visibility is actually an AZ::u32 enum class, so we need
+                                // to check here if we read in a 0 or 1 instead of a hash so we can handle
+                                // those special cases.
+                                AZ::u32 visibilityNumericValue = static_cast<AZ::u32>(visibility);
+                                if (visibilityNumericValue == 0)
+                                {
+                                    visibility = PropertyVisibility::Hide;
+                                }
+                                else if (visibilityNumericValue == 1)
+                                {
+                                    visibility = PropertyVisibility::Show;
+                                }
+                                return;
+                            }
+                            else
+                            {
+                                auto visibilityBoolValue = VisibilityBoolean
+                                    .DomToValue(VisibilityBoolean.LegacyAttributeToDomValue(instance, it->second));
+                                if (visibilityBoolValue.has_value())
+                                {
+                                    bool isVisible = visibilityBoolValue.value();
+                                    visibility = isVisible ? PropertyVisibility::Show : PropertyVisibility::Hide;
+                                    return;
+                                }
+                            }
```

Comment:
nit: Since the previous if always returns, we could remove the else and avoid the indentation.

Also, initialize the auto inside the if:
```
if (auto visibilityBoolValue = VisibilityBoolean.DomToValue(VisibilityBoolean.LegacyAttributeToDomValue(instance, it->second)); visibilityBoolValue.has_value())
{
    bool isVisible = visibilityBoolValue.value();
    visibility = isVisible ? PropertyVisibility::Show : PropertyVisibility::Hide;
    return;
}
```

or even better, if `visibilityBoolValue` can implicitly convert to bool:

```
if (auto visibilityBoolValue = VisibilityBoolean.DomToValue(VisibilityBoolean.LegacyAttributeToDomValue(instance, it->second)))
{
// ...
```

Reply:
Good call I'll clean up!